src/nupic/algorithms/backtracking_tm.py

```diff
@@ -1355,7 +1357,7 @@ def getSegmentOnCell(self, c, i, segIdx):
     return retlist
 
 
-  class _SegmentUpdate(Serializable):
+  class _SegmentUpdate(object):
```

Comment:
Why don't we want `Serializable` base class here any longer? Doesn't it break our goal of having all such classes based on `Serializable`?

I can see that you changed the methods (+ additional arg), so that they are not compatible with `Serializable` interface. Does this mean that `Serializable` is not sufficiently flexible for our use cases?

Same in `Segment` class...